File: treatment/src/main/java/com/hartwig/actin/treatment/database/config/InclusionCriteriaConfigFactory.java
package com.hartwig.actin.treatment.database.config;

import java.util.Map;

import com.hartwig.actin.treatment.database.TreatmentDatabaseUtil;
import com.hartwig.actin.treatment.datamodel.EligibilityRule;

import org.jetbrains.annotations.NotNull;

public class InclusionCriteriaConfigFactory implements TrialConfigFactory<InclusionCriteriaConfig> {

    @NotNull
    @Override
    public InclusionCriteriaConfig create(@NotNull Map<String, Integer> fields, @NotNull String[] parts) {
        return ImmutableInclusionCriteriaConfig.builder()
                .trialId(parts[fields.get("trialId")])
                .appliesToCohorts(TreatmentDatabaseUtil.toCohorts(parts[fields.get("appliesToCohorts")]))
                .eligibilityRule(EligibilityRule.valueOf(parts[fields.get("eligibilityRule")]))
                .eligibilityParameters(TreatmentDatabaseUtil.toParameters(parts[fields.get("eligibilityParameters")]))
                .build();
    }
}

File: trial/src/main/kotlin/com/hartwig/actin/trial/trial/config/InclusionCriteriaConfigFactory.kt
package com.hartwig.actin.trial.trial.config

import com.hartwig.actin.trial.trial.TrialConfigDatabaseUtil

class InclusionCriteriaConfigFactory : TrialConfigFactory<InclusionCriteriaConfig> {
    override fun create(fields: Map<String, Int>, parts: List<String>): InclusionCriteriaConfig {
        return InclusionCriteriaConfig(
            trialId = parts[fields["trialId"]!!],
            referenceIds = TrialConfigDatabaseUtil.toReferenceIds(parts[fields["referenceIds"]!!]),
            appliesToCohorts = TrialConfigDatabaseUtil.toCohorts(parts[fields["appliesToCohorts"]!!]),
            inclusionRule = parts[fields["inclusionRule"]!!]
        )
    }
}